osf/management/commands/sync_doi_metadata.py

```diff
@@ -10,32 +11,45 @@
 logger = logging.getLogger(__name__)
 
 
-@app.task(name='osf.management.commands.sync_doi_metadata', max_retries=5, default_retry_delay=60)
-def sync_identifier_doi(identifier_id):
+@app.task(name='osf.management.commands.sync_doi_metadata', max_retries=5, default_retry_delay=60 * 5)
+def sync_identifier_doi(identifier_id, delay=0):
+    time.sleep(delay)
+
     identifier = Identifier.objects.get(id=identifier_id)
     identifier.referent.request_identifier_update('doi')
     identifier.save()
     logger.info(f' doi update for {identifier.value} complete')
 
 
-def sync_doi_metadata(modified_date, batch_size=100, dry_run=True, sync_private=False):
+def sync_doi_metadata(modified_date, batch_size=100, dry_run=True, sync_private=False, rate_limit=100):
 
     identifiers = Identifier.objects.filter(
         category='doi',
         deleted__isnull=True,
         modified__lte=modified_date,
         object_id__isnull=False,
-    )[:batch_size]
+    )
+    if batch_size:
+        identifiers = identifiers[:batch_size]
+        rate_limit = batch_size if batch_size > rate_limit else rate_limit
+
     logger.info(f'{"[DRY RUN]: " if dry_run else ""}'
                 f'{identifiers.count()} identifiers to mint')
 
-    for identifier in identifiers:
-        if not dry_run:
-            if (identifier.referent.is_public and not identifier.referent.deleted and not identifier.referent.is_retracted) or sync_private:
-                sync_identifier_doi.apply_async(kwargs={'identifier_id': identifier.id})
+    delay = 0
+    for record_number, identifier in enumerate(identifiers, 1):
+        if dry_run:
+            logger.info(f'{"[DRY RUN]: " if dry_run else ""}'
+                        f' doi minting for {identifier.value} started')
+            continue
 
-        logger.info(f'{"[DRY RUN]: " if dry_run else ""}'
-                    f' doi minting for {identifier.value} started')
+        # in order to not reach rate limits that CrossRef and DataCite have, we increase delay after each
+        # rate_limit records by 5 minutes
+        if not record_number % rate_limit:
+            delay += 60 * 5
```

Comment:
Probably this (and the `default_retry_delay`) should be set from a constant at the top of the file